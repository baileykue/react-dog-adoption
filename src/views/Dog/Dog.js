import React, { useState, useEffect } from 'react';
import { useParams, useHistory } from 'react-router-dom';
import DogDetail from '../../components/DogDetail/DogDetail';
import { deleteDogById, fetchById } from '../../services/dogs';

export default function Dog() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchById(params.id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  const handleEdit = async () => {
    history.push(`/dogs/${params.id}/edit`);
  };

  const handleDelete = async () => {
    await deleteDogById(params.id);
    history.push('/');
  };

  if (loading) return <h3>Loading doggy info...</h3>;

  return (
    <div>
      <DogDetail {...dog} handleEdit={handleEdit} handleDelete={handleDelete} showDetail />
    </div>
  );
}
