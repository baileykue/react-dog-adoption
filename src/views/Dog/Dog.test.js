import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Dog from './Dog';

test('<Dog/> should render the specific dog information', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/4']}>
      <Route path="/dogs/:id">
        <Dog />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Franny');
  expect(container).toMatchSnapshot();
});
