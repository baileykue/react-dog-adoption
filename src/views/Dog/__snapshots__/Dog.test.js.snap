// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dog/> should render the specific dog information 1`] = `
<div>
  <div>
    <div
      class="detail"
    >
      <img
        class="bio-img"
        src="https://placedog.net/500?id=51"
      />
      <h2>
        Franny
      </h2>
      <p>
        Sweet Frenchie with lots of love to give.
      </p>
    </div>
  </div>
</div>
`;
